feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go

```diff
@@ -282,36 +288,99 @@ func TestHardwarecards(t *testing.T) {
 	}
 }
 
+func findComponentsListByType(t *testing.T, dut *ondatra.DUTDevice) map[string][]string {
+	t.Helper()
+	componentType := map[string]oc.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT{
+		"Chassis":     chassisType,
+		"Fabric":      fabricType,
+		"Linecard":    linecardType,
+		"PowerSupply": powerSupplyType,
+		"Supervisor":  supervisorType,
+		"SwitchChip":  switchChipType,
+		"Transceiver": transceiverType,
+		"Fan":         fanType,
+		"FabricChip":  switchChipType,
+		"TempSensor":  sensorType,
+	}
+	if len(componentsByType) != 0 {
+		return componentsByType
+	}
+	componentsByType = make(map[string][]string)
+	components := gnmi.GetAll(t, dut, gnmi.OC().ComponentAny().State())
+	for compName := range componentType {
+		for _, c := range components {
+			if strings.Contains(c.GetName(), "Fan") {
+				t.Logf("Component %s type is missing from telemetry", c.GetName())
+			}
+
+			if c.GetType() == nil {
+				t.Logf("Component %s type is missing from telemetry", c.GetName())
+				continue
+			}
+			t.Logf("Component %s has type: %v", c.GetName(), c.GetType())
+			if v := c.GetType(); v == componentType[compName] {
```

Comment:
Why not just "c.GetType() == componentType[compName]"?

Reply:
fixed